MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Reflectometry/DataPreprocessorAlgorithm.h

```
@@ -0,0 +1,135 @@
+
+Copyright &copy; 2011-14 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>.
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+class DataPreprocessorAlgorithm {
+public:
+  /** Constructor
+  * @param name : The name of the pre-processing algorithm
+  * @param prefix : The prefix that will added to the output workspace name
+  * @param blacklist : The list of properties we don't want to show
+  * @param show : Whether or not to show the information associated with
+  * this pre-processor in the processed workspace's name
+  */
+  DataPreprocessorAlgorithm(
+      const std::string &name, const std::string &prefix = "",
+      const std::set<std::string> &blacklist = std::set<std::string>(),
+      bool show = true)
+      : m_name(name), m_prefix(prefix), m_blacklist(blacklist), m_show(show) {
+
+    Mantid::API::IAlgorithm_sptr alg =
+        Mantid::API::AlgorithmManager::Instance().create(m_name);
+
+    int countInputWS = 0;
+    int countOutputWS = 0;
+
+    auto properties = alg->getProperties();
+    for (auto &prop : properties) {
+
+      if (prop->direction() == 0 &&
+          (prop->type() == "MatrixWorkspace" || prop->type() == "Workspace")) {
```

Again, issue about swallowing errors.